subsys/bluetooth/services/fast_pair/fp_gatt_service.c

```diff
@@ -12,6 +12,12 @@
 #include <zephyr/bluetooth/l2cap.h>
 #include <zephyr/bluetooth/gatt.h>
 
+/* Those headers are included to make a workaround of Android issue with sending old RPA address
+ * during Personalized Name write.
+ */
+#include <host/hci_core.h>
```

Comment:
What part of this internal API do you use? (I see the usage of the ```common/rpa``` header but I don't see the usage of this one)

Reply:
In line 306 I use `bt_dev` variable which is defined in `host/hci_core.c` and externed to `host/hci_core.h`. We need this to get our own IRK.

`/* This is a workaround of Android issue with sending old RPA address during
		 * Personalized Name write.
		 */
		if (!bt_rpa_irk_matches(bt_dev.irk[conn_info.id],
					(const bt_addr_t *)parsed_req->provider_address)) {
			return -EINVAL;
		}`